jobs/ci-run/integration/gen/test-relations.yml

```
@@ -106,11 +106,22 @@
       - select-oci-registry
       - wait-for-cloud-init
       - prepare-integration-test
-      - run-integration-test:
-            test_name: 'relations'
-            setup_steps: ''
-            task_name: 'test_relation_data_exchange'
-            skip_tasks: 'test_relation_departing_unit,test_relation_list_app,test_relation_model_get'
+      - conditional-step:
+          # Do not run on regexp version match.
+          # Accounts for tests which do not exist
+          # in later Juju versions.
+          condition-kind: not
+          condition-operand:
+            condition-kind: regex-match
+            regex: "^[5-9].*|^4\\.([0-9]|\\d{{2,}})(\\.|-).*"
```

@jack-w-shaw  This generated regexp looks suspicious to me. Why do we have double braces around the `\d` specifier ? 

(it comes from this code you initiate a year ago ;) 
https://github.com/juju/juju-qa-jenkins/blob/bd9934774baa13b1a3855d4109e1dd69a02462c4/tools/gen-wire-tests/main.go#L70-L79)

It should be:
```suggestion
            regex: "^[5-9].*|^4\\.([0-9]|\\d{2,})(\\.|-).*"
```

Which is in fact totally equivalent to (since `\d` <=> `[0-9]` and so if i have 1 or at least two... it is at least one, which is `+`

```suggestion
            regex: "^[5-9].*|^4\\.\\d+(\\.|-).*"
```

What do I miss ?

---

Digging more - thanks @jack-w-shaw ;) 

Those bracket have been added will moving to noble:

https://github.com/juju/juju-qa-jenkins/pull/196/files#diff-15cc5fb476a56c5b55c6cc61bc3719d2c4493656cb4d12bca7c948e224f33797R71-R78

@hpidcock: any idea ?

---

You'll have to ask @hpidcock about the double-bracketting

I would be opposed to replacing `[0-9]` with `\d` in these places, because the lines below we have `[1-9]`, `[2-9]`, etc., so `[0-9]` is much more readable line-to-line

---

`{var}` is a jenkins job builder template variable substitution
`{{}}` is escaped by jenkins job builder to be `{}`